### PR TITLE
Stencil split toolbox helper

### PR DESCRIPTION
**Pull request by cboesch-gpsw**

### CODE REVIEW

File: pynodegl-utils/pynodegl_utils/toolbox/scenes.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        scene_mask,
+        color_write_mask="",  # no need to write the color themselves, only the stencil
+        stencil_test=True,
+        stencil_ref=1,
+        stencil_read_mask=0,
+        stencil_write_mask=0xFF,
+        stencil_fail="replace",
+        stencil_depth_fail="replace",
+        stencil_depth_pass="replace",
+    )
```
Comment (mbouron-gpsw) [inline pynodegl-utils/pynodegl_utils/toolbox/scenes.py:41]: You may want to clear the whole stencil buffer with 0 so if the underlying scene use it, it does not interfere with the compare method.

**Comment by mbouron-gpsw** [comment]
`because we are doing 2 draws instead of one)`: it is actually 3 draws (4 if you clear the whole stencil buffer with 0)

File: pynodegl-utils/pynodegl_utils/toolbox/scenes.py (hunk 1 above)
Comment (mbouron-gpsw) [reply pynodegl-utils/pynodegl_utils/toolbox/scenes.py:41]: Or state the limitation regarding the stencil usage in the documentation of the function.

Comment (cboesch-gpsw) [reply pynodegl-utils/pynodegl_utils/toolbox/scenes.py:45]: Done

**Comment by cboesch-gpsw** [comment]
> `because we are doing 2 draws instead of one)`: it is actually 3 draws (4 if you clear the whole stencil buffer with 0)

Message adjusted

**Review by mbouron-gpsw — APPROVED** [review]

**Pull request merged**